src/controllers/category.controller.js

```diff
@@ -0,0 +1,48 @@
+import categoryService from "../services/category.service.js";
+
+const ITEMS_PER_PAGE = 10; // Số bài viết trên mỗi trang
+
+export const getCategoryArticles = async (req, res) => {
+    try {
+        const categoryId = req.query.id;
+        const page = parseInt(req.query.page) || 1;
+
+        if (!categoryId) {
+            return res.status(400).render('../views/vwError/404.ejs');
```

Comment:
Did you test this? It can be simplified to `res.redirect("/404")`

Reply:
Okay, I gonna changed it rn